### DISCUSSION TITLE
Bug: Wiki Search Results Showing Deleted Pages

### DISCUSSION THREAD

**Discussion by cloaked2222** [community/community#22085] [New to GitHub]
<p>When searching a wiki, deleted pages are included in the search results.</p>
<p>To reproduce this bug:</p>
<ol>
<li>Create a page named “Test 1”</li>
<li>Delete the page (or rename the page from “Test 1” to “Test 2”).</li>
<li>Search for “Test”, then select the “Wikis” tab.</li>
<li>“Test 1” (and “Test 2”) will usually appear in the search results.</li>
<li>When a user clicks “Test 1”, they will be taken to the “Create new page”. This is very confusing for users. Users are not notified that the page has been deleted, and users aren’t shown the version history of the deleted page.</li>
</ol>
<p>Clearly this is not good. I have a wiki that contains 1,000+ pages. A few weeks ago we renamed all the pages to a new naming convention, and now weeks later search results are still showing most of these deleted pages alongside the current pages… This has rendered search not user friendly. When searching the contents of a wiki, search results should not show deleted pages.</p>

**Comment by yamiacat** [comment]
<p>Hi <a class="mention" href="/cloaked2222">@cloaked2222</a> - I haven’t been able to replicate this so far - when I remove pages from my test wikis they’re also removed from search results within a few seconds.</p>
<p>Can you let me know which repository this is happening on, and I’ll trigger a re-index of the wiki to see if that helps!</p>

**Reply by elenabaurkot** [reply]
@yamiacat how do you trigger a re-index of a wiki, I have been editing a wiki and am experiencing this exact issue where deleted pages are showing up in my searches and when I click them they are taking me to "Create new page", not sure how to actually get rid of those pages

**Reply by yamiacat** [reply]
Hi @elenabaurkot - you'll need to contact GitHub Support to do that, I don't believe you can do it from your side!

If it's not many pages it might be easier to solve it as one of the other users in this thread did - recreate the page in question, and then delete it again.

**Reply by elenabaurkot** [reply]
Thanks for the reply @yamiacat, I will give that a try now and contact support if I can't figure it out!

**Comment by tajmone** [comment]
<aside class="quote group-github-employees" data-username="yamiacat" data-post="2" data-topic="120001">
<div class="title">
<div class="quote-controls"></div>
 yamiacat:</div>
<blockquote>
<p>I haven’t been able to replicate this so far</p>
</blockquote>
</aside>
<p>I wonder if the problem could be cache-related. In the past I’ve experienced website refresh issues due to ISPs’ caching — had to use a proxy to see the updates in real time, even with dynamically generated pages, because my ISP apparently was showing me cached results.  Either that or local caching could be the reasons why old results show up in searches?</p>

**Comment by yamiacat** [comment]
<p>I think it’s definitely cache related, yes!  I have now been able to replicate it once or twice. Reloading the page will sometimes get deleted items to reappear. I’ll create an internal issue to look into this.</p>

**Comment by yamiacat** [comment]
<p>I’m glad you got it working correctly again <a class="mention" href="/cloaked2222">@cloaked2222</a> - sorry that it was such a hassle though!</p>

**Comment by cloaked2222** [comment, marked as answer]
<p>I experienced this bug on several repositories in late May/June. However as of 6/29, this issue appears to have been resolved for newly deleted pages. I also manually fixed most previously deleted pages, by recreating the pages, and then deleting them again.</p>
<p>I don’t think this bug was cache related. On <a href="https://github.com/HWRM/KarosGraveyard/wiki" rel="nofollow noopener">this</a> wiki, in late May/June I renamed all ~1000 pages. Almost all pages contained the text “Comments”. Doing a search on “Comments” yielded about 1500 results, including ~1000 pages and ~500 deleted pages. This search results bug persisted for weeks even on brand new searches until 6/29 when I was able to resolve it by recreating each page, and then deleting them again.</p>

**Comment by dscataglini** [comment]
This happens within my organization as well. I think it's kinda ridiculous that it still hasn't fixed and the only "fix" requires to re-add all the (deleted) files and then deleting them again.
This should be addressed. It seems the indexing of the wiki has bugs, and should re-index on delete files
